indexing-service/src/main/java/org/apache/druid/indexing/common/ParallelCompactionTaskReportData.java

```diff
@@ -0,0 +1,113 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.druid.indexing.common;
+
+import com.fasterxml.jackson.annotation.JsonProperty;
+import org.apache.druid.indexer.IngestionState;
+
+import javax.annotation.Nullable;
+import java.util.Map;
+import java.util.Objects;
+
+/**
+ * This class has fields specific to the parallel compaction task and hence extends the generic
+ * task report data {@link IngestionStatsAndErrorsTaskReportData}
+ */
+public class ParallelCompactionTaskReportData extends IngestionStatsAndErrorsTaskReportData
```

Comment:
Why do we need a new class for just 2 new fields?
`segmentsPublished` and `segmentsRead` is information that can be useful in the case of non-compaction jobs too. We can use the existing `IngestionStatsAndErrorsTaskReportData` class and add these two as nullable fields.

Reply:
https://github.com/apache/druid/pull/15947#pullrequestreview-1909752276

Basically to avoid confusion of being null where these fields still mean something but haven't got functionality yet to populate them.

Reply:
The way to do that would be to use the annotation `@JsonInclude(JsonInclude.Include.NON_NULL)` so that only non-null values get serialized.

Reply:
Done.